### PR TITLE
Task status managment

### PR DESCRIPTION
**Pull request by nzlatev7**

### CODE REVIEW

File: TaskManagementSystem/Services/TasksService.cs [hunk 1]
```diff
@@ -56,27 +61,35 @@ public async Task<TaskResponseDto> GetTaskByIdAsync(int taskId)
...
 
         if (taskEntity is null)
-            throw new NotFoundException(ValidationMessages.TaskDoesNotExist);
+            throw new NotFoundException(ErrorMessageConstants.TaskDoesNotExist);
 
         return taskEntity.ToOutDto();
     }
 
-    public async Task<TaskResponseDto?> UpdateTaskAsync(int taskId, TaskRequestDto taskDto)
+    public async Task<TaskResponseDto?> UpdateTaskAsync(int taskId, UpdateTaskRequestDto taskDto)
```
Comment (BoykoBonev) [inline TaskManagementSystem/Services/TasksService.cs:69]: why taskId is duplicated outside taskDto?

File: TaskManagementSystem/Services/TasksService.cs [hunk 2]
```diff
@@ -56,27 +61,35 @@ public async Task<TaskResponseDto> GetTaskByIdAsync(int taskId)
...
     {
         var taskEntity = await _dbContext.Tasks.FindAsync(taskId);
 
         if (taskEntity is null)
-            throw new NotFoundException(ValidationMessages.TaskDoesNotExist);
+            throw new NotFoundException(ErrorMessageConstants.TaskDoesNotExist);
```
Comment (BoykoBonev) [inline TaskManagementSystem/Services/TasksService.cs:74]: extract the validation logic

File: TaskManagementSystem/Services/TasksService.cs [hunk 3]
```diff
@@ -56,27 +61,35 @@ public async Task<TaskResponseDto> GetTaskByIdAsync(int taskId)
...
+
+        if (taskEntity.Status != Status.Completed && taskDto.Status == Status.Archived)
+            throw new BadHttpRequestException(ErrorMessageConstants.OnlyCompletedTaskCanBeArchived);
+
+        if (taskEntity.Status == Status.Archived && taskDto.Status != Status.Archived)
```
Comment (BoykoBonev) [inline TaskManagementSystem/Services/TasksService.cs:79]: Should we allow archived tasks to be modified?

File: TaskManagementSystem/Services/TasksService.cs [hunk 4]
```diff
@@ -91,6 +104,14 @@ public async Task DeleteTaskAsync(int taskId)
             .ExecuteDeleteAsync();
 
         if (deletedRows is 0)
-            throw new NotFoundException(ValidationMessages.TaskDoesNotExist);
+            throw new NotFoundException(ErrorMessageConstants.TaskDoesNotExist);
+    }
+
+    private bool IsCompleted(Status taskStatus)
+    {
```
Comment (BoykoBonev) [inline TaskManagementSystem/Services/TasksService.cs:111]: single return

File: TaskManagementSystem/Services/TasksService.cs (hunk 1 above)
Comment (nzlatev7) [reply TaskManagementSystem/Services/TasksService.cs:69]: The taskId is part of the URL while taskDto is body parameter, maybe we can introduce single DTO which unites them

**Pull request merged**